Python/Product/Analysis/AnalysisUnit.cs

```diff
@@ -421,8 +423,51 @@ public ClassAnalysisUnit(ClassDefinition node, InterpreterScope declScope, Analy
                 }
             }
 
+
             ddg.SetCurrentUnit(this);
             ddg.WalkBody(Ast.Body, classInfo.AnalysisUnit);
+
+            ddg.SetCurrentUnit(_outerUnit);
+            var v = _outerUnit.Scope.AddLocatedVariable(Ast.Name, Ast.NameExpression, this);
+            v.AddTypes(this, ProcessClassDecorators(ddg, classInfo));
+        }
+
+        private IAnalysisSet ProcessClassDecorators(DDG ddg, ClassInfo classInfo) {
+            var types = classInfo.SelfSet;
+            if (Ast.Decorators != null) {
+                Expression expr = Ast.NameExpression;
+
+                foreach (var d in Ast.Decorators.DecoratorsInternal) {
```

Comment:
`Ast.Decorators.DecoratorsInternal.Where`(x => x != null)?

Reply:
Alex changed all of these to use the arrays directly for speed. Fairly sure adding Linq here will remove all of that benefit